src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -202,7 +202,17 @@ internal static string ReadBufferAsString(ArraySegment<byte> buffer, HttpContent
             {
                 try
                 {
-                    encoding = Encoding.GetEncoding(headers.ContentType.CharSet);
+                    // Remove at most a single set of quotes.
+                    if (headers.ContentType.CharSet.Length > 2 &&
```

Comment:
Nit: how about storing `headers.ContentType` and `headers.ContentType.CharSet` into locals?  They don't look particularly cheap to access, and we're doing it in this block (and the earlier if)

Reply:
Thanks for the feedback. This should be addressed in the latest commit.